coral/src/app/features/approvals/topics/TopicApprovals.tsx

```diff
@@ -2,79 +2,39 @@ import { NativeSelect, SearchInput } from "@aivenio/aquarium";
 import { ApprovalsLayout } from "src/app/features/approvals/components/ApprovalsLayout";
 import { Pagination } from "src/app/components/Pagination";
 import { TopicApprovalsTable } from "src/app/features/approvals/topics/components/TopicApprovalsTable";
-import { TopicRequest } from "src/domain/topic";
+import { useQuery } from "@tanstack/react-query";
+import { getTopicRequestsForApprover } from "src/domain/topic/topic-api";
+import { useSearchParams } from "react-router-dom";
 
-const mockedRequests: TopicRequest[] = [
-  {
-    topicname: "test-topic-1",
-    environment: "1",
-    topicpartitions: 4,
-    teamname: "NCC1701D",
-    remarks: "asap",
-    description: "This topic is for test",
-    replicationfactor: "2",
-    environmentName: "BRG",
-    topicid: 1000,
-    advancedTopicConfigEntries: [
-      {
-        configKey: "cleanup.policy",
-        configValue: "delete",
-      },
-    ],
-    requestOperationType: "CREATE",
-    requestor: "jlpicard",
-    requesttime: "1987-09-28T13:37:00.001+00:00",
-    requesttimestring: "28-Sep-1987 13:37:00",
-    requestStatus: "CREATED",
-    totalNoPages: "1",
-    approvingTeamDetails:
-      "Team : NCC1701D, Users : jlpicard, worf, bcrusher, geordilf,",
-    teamId: 1003,
-    allPageNos: ["1"],
-    currentPage: "1",
-    deletable: true,
-    editable: true,
-  },
-  {
-    topicname: "test-topic-2",
-    environment: "1",
-    topicpartitions: 4,
-    teamname: "MIRRORUNIVERSE",
-    remarks: "asap",
-    description: "This topic is for test",
-    replicationfactor: "2",
-    environmentName: "SBY",
-    topicid: 1001,
-    advancedTopicConfigEntries: [
-      {
-        configKey: "compression.type",
-        configValue: "snappy",
-      },
-    ],
+function TopicApprovals() {
+  const [searchParams, setSearchParams] = useSearchParams();
 
-    requestOperationType: "UPDATE",
-    requestor: "bcrusher",
-    requesttime: "1994-23-05T13:37:00.001+00:00",
-    requesttimestring: "23-May-1994 13:37:00",
-    requestStatus: "APPROVED",
-    totalNoPages: "1",
-    approvingTeamDetails:
-      "Team : NCC1701D, Users : jlpicard, worf, bcrusher, geordilf,",
-    teamId: 1003,
-    allPageNos: ["1"],
-    currentPage: "1",
-    deletable: true,
-    editable: true,
-  },
-];
+  const initialPage = searchParams.get("page")
```

Comment:
I don't think this is `initialPage`, it's `currentPage`, isn't it? As it will be updated every time we set the search params?

Reply:
True, that's a better naming!